### PR TITLE
SDN-4434: Network diagnostics scheduling

### PR DESCRIPTION
**Pull request by kyrtapz**
Implements: https://github.com/openshift/enhancements/pull/1585

Note: The API is currently in TechPreview, the plan is to enable it in GA in the current release.

### CODE REVIEW

**Comment by kyrtapz** [comment]
/hold
waiting for a review from @bpickard22 and @trozet

File: pkg/network/render.go
Comment (trozet) [inline pkg/network/render.go:808]: if Mode is empty, where are you defaulting it?

File: pkg/network/render.go
Comment (kyrtapz) [reply pkg/network/render.go:110]: We do not follow that pattern here and the `networkDiagnostics` is configured through clusterConf so we don't need to copy it over. operConf has the legacy `disableNetworkDiagnostics` field which we fallback to and the plan is to deprecate it.

File: pkg/controller/operconfig/operconfig_controller.go
Comment (kyrtapz) [reply pkg/controller/operconfig/operconfig_controller.go:135]: This controller doesn't read other config fields from `configv1.Network`.

File: pkg/network/render.go
Comment (kyrtapz) [reply pkg/network/render.go:808]: If the mode is empty and `operConf.DisableNetworkDiagnostics` is not set then we render the diagnostics.
Empty is an allowed value in the API, from the API docs:
```
 this means the user has no opinion and the platform is left
 to choose reasonable defaults. These defaults are subject to change over time.
 The current default is All.
```

**Comment by kyrtapz** [comment]
Had to do a manual rebase because even though the automatic one was successful the code did not build.
The change I had to make:
```
--- a/pkg/network/render.go
+++ b/pkg/network/render.go
@@ -132,7 +132,7 @@ func Render(operConf *operv1.NetworkSpec, clusterConf *configv1.NetworkSpec, man
        }
        objs = append(objs, o...)
 
-       o, err = renderIPTablesAlerter(conf, bootstrapResult, manifestDir)
+       o, err = renderIPTablesAlerter(operConf, bootstrapResult, manifestDir)
        if err != nil {
                return nil, progressing, err
        }

```

**Review by trozet — APPROVED** [review]
/lgtm

**Pull request merged**
